react/AccordionItem/AccordionItem.js

```diff
@@ -0,0 +1,102 @@
+import React, { Fragment, useRef, useState } from 'react';
+import PropTypes from 'prop-types';
+import { ChevronIcon } from 'seek-style-guide/react';
+import classnames from 'classnames';
+import styles from './AccordionItem.less';
+import { toggleContent } from './utils';
+import { CLOSED_HEIGHT } from './constants';
+
+function AccordionItem({
+  className,
+  title,
+  children,
+  open,
+  onOpen,
+  onClose,
+  ...restProps
+}) {
+  const initialHeight = open ? 'auto' : CLOSED_HEIGHT;
+  const initialVisibility = open ? 'visible' : 'hidden';
+  const initialOverflow = initialVisibility;
+  const contentEl = useRef(null);
+  const [currentHeight, setCurrentHeight] = useState(initialHeight);
+  const [cssVisibility, setCssVisibility] = useState(initialVisibility);
+  const [cssOverflow, setCssOverflow] = useState(initialOverflow);
+  const [timeoutHandle, setTimeoutHandle] = useState(null);
+  const [isOpen, setIsOpen] = useState(open);
+
+  const buttonClasses = classnames(className, styles.title);
+
+  const isAnimating = currentHeight === CLOSED_HEIGHT && isOpen;
+  const expanderClasses = classnames(styles.expander, {
+    [styles.expanderOpen]: isAnimating
+  });
+  const chevronDirection = isOpen ? 'up' : 'down';
+
+  return (
+    <Fragment>
+      <button
+        type="button"
+        className={buttonClasses}
+        onClick={() => {
+          toggleContent({
+            el: contentEl.current,
+            setCurrentHeight,
+            timeoutHandle,
+            setTimeoutHandle,
+            isOpen,
+            setIsOpen,
+            setCssVisibility,
+            setCssOverflow
+          });
+
+          if (isOpen && onClose) {
+            onClose();
+          } else if (!isOpen && onOpen) {
+            onOpen();
+          }
+        }}
+        {...restProps}
+      >
+        {typeof title === 'string' ? (
+          <span className={styles.titleText}>{title}</span>
```

Comment:
Could this leverage the `<Text />` or `<TextLink />` components rather than use it's own styles?

Reply:
I did consider that, but I'd prefer not to nest an anchor tag inside of a button?

Reply:
As discussed, have also tried replacing the button element with `<Button transparent />`, but it takes on the "bounce" effect, and other styling related to that component. Will be simpler to keep current styling rather than writing overrides.